mpv/mpv.go

```diff
@@ -44,20 +51,31 @@ type Request struct {
 }
 
 type Client struct {
-	outgoing          chan<- []byte
-	conn              *connection
-	mu                sync.Mutex
-	paused            bool
-	playbackRestarted bool
+	outgoing     chan<- []byte
+	conn         *connection
+	mu           sync.Mutex
+	paused       bool
+	playbackTime float64
+	syncMargin   float64
 }
 
-func (s *Client) handleState(event Event) {
+func (s *Client) handleState(event Event) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	switch event.Name {
 	case eventPause:
 		s.paused = event.Data == "yes"
+	case eventPlaybackTime:
+		playbackTime, err := strconv.ParseFloat(event.Data, 64)
+		if err != nil {
+			return err
+		}
+		playbackTime = math.Floor(playbackTime)
+		if s.playbackTime == playbackTime {
+			return errNoChange
+		}
 	}
+	return nil
```

Comment:
Can you elaborate what are you doing here?

Reply:
yes i can

Reply:
we get the current playback time from the player, and we compare it to the latest playback if it's within the same second we skip this event. this to keep track of the current second we're playing.
actually setting `s.playbackTime = playbackTime` is missing